google/cloud/spanner_dbapi/cursor.py

```diff
@@ -258,9 +258,26 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
-        for params in seq_of_params:
-            self.execute(operation, params)
-            many_result_set.add_iter(self._itr)
+        if classification in (parse_utils.STMT_INSERT, parse_utils.STMT_UPDATING):
```

Comment:
I think we should skip the parsing here and assume that the method is only called with DML statements. The DbApi specification states that implementations are free to accept or refuse statements that would return result sets. Cloud Spanner will refuse queries in the `ExecuteBatchDml` RPC, so we can leave that check to the backend. That means that this implementation will only support DML in `executemany`, but that is not an unreasonable limitation.